web3/middleware/stalecheck.py

```diff
@@ -0,0 +1,44 @@
+import time
+
+from web3.exceptions import StaleBlockchain
+
+SKIP_STALECHECK_FOR_METHODS = set([
+    'eth_getBlockByNumber',
+])
+
+
+def _isfresh(block, allowable_delay):
+    return block and time.time() - block['timestamp'] <= allowable_delay
+
+
+def make_stalecheck_middleware(allowable_delay=0):
+    '''
+    Use to require that a function will run only of the blockchain is recently updated.
+
+    This middleware takes an argument, so unlike other middleware, you must make the middleware
+    with a method call.
+    For example: `make_stalecheck_middleware(allowable_delay=60*5)`
+
+    If the latest block in the chain is older than 5 minutes in this example, then the
+    middleware will raise a StaleBlockchain exception.
+    '''
+    if not allowable_delay:
```

Comment:
This seems like the check should be `if delay <= 0:` since the delay should also be strictly positive as well as non-zero.